### DISCUSSION TITLE
Indexed DB - Attempt to Iterate a cursor that doesn't exist

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#484] [Q&A]
I have an error that comes up when using the indexed_db from the Storage module.
The error doesn't appear consistently but does seem to be when an iPhone locks or the app is sent minimised.

The code that is logging the error is actually running in a anvil labs web worker and is nothing more than a `store.get()`.

Handling this error then creates another error at the same line which is "Attempt to get a record from database without an in-progress transaction".

I suspect this is something to do with the app being paused by Safari as I cannot recreate this on Chrome.

Is there any advice on how to handle this?

**Comment by s-cork** [comment]
Happy to have a look if you can provide a repo. I have macs and iPhones so can test it out.

**Reply by rhurlbatt** [reply]
Thanks Stu. I have built a minimal clone based on the Fib example.
Web worker code below:
```python
from anvil_extras.storage import indexed_db
store = indexed_db.create_store("results")

def fib(num):
    store.clear()
    a, b = 1, 0
    i = 0
    print(worker) # worker is a global object injected into worker module
    while i < num:
        a, b = a + b, a
        i += 1
        worker.task_state["i"] = i
        
        i_minus_1 = i-1

        # Attempt to throw indexed_db error from get
        previous_row = store.get(str(i_minus_1))
        
        worker.task_state["i_minus_1"] = i_minus_1

        # Attempt to throw indexed_db error from storing
        store[str(i)] = a
    return b
```
To recreate run the calc and then lock the iPhone. After a while, I presume the indexed_db goes to sleep and then the web work fails at the `previous_row = store.get(str(i_minus_1))` line (I think - I don't have a Mac to access Safari dev tools).

If you remove this line, it will fail with another error on the `store[str(i)] = a` line.

[fib-web-worker.anvil.app](fib-web-worker.anvil.app)

[Clone Link](https://anvil.works/build#clone:45OROQNNY77JQACC=UZCHMG5NNA4I2OD5H47UGRRZ)

**Comment by rhurlbatt** [comment]
Some more info on similar issues:
[https://github.com/pouchdb/pouchdb/issues/8329](https://github.com/pouchdb/pouchdb/issues/8329)
[https://github.com/firebase/firebase-js-sdk/issues/2232](https://github.com/firebase/firebase-js-sdk/issues/2232)
